react/features/video-layout/functions.js

```diff
@@ -35,30 +35,49 @@ export function getMaxColumnCount() {
 }
 
 /**
- * Returns the cell count dimensions for tile view. Tile view tries to uphold
- * equal count of tiles for height and width, until maxColumn is reached in
- * which rows will be added but no more columns.
+ * Returns the cell count dimensions for tile view. Tile view tries to fill
+ * the available space optimally
  *
  * @param {Object} state - The redux store state.
- * @param {number} maxColumns - The maximum number of columns that can be
- * displayed.
  * @returns {Object} An object is return with the desired number of columns,
  * rows, and visible rows (the rest should overflow) for the tile view layout.
  */
-export function getTileViewGridDimensions(state: Object, maxColumns: number = getMaxColumnCount()) {
+export function getTileViewGridDimensions(state: Object) {
     // When in tile view mode, we must discount ourselves (the local participant) because our
     // tile is not visible.
     const { iAmRecorder } = state['features/base/config'];
     const numberOfParticipants = state['features/base/participants'].length - (iAmRecorder ? 1 : 0);
+    const { clientHeight, clientWidth } = state['features/base/responsive-ui'];
+    const aspectRatio = state['features/base/responsive-ui'].aspectRatio;
```

Comment:
you can extract this too in the above line

Reply:
Coding javascript feels like throwing spaghetti at the wall and seeing what sticks. Guesstimated a thing that somehow might magically work. No clue.

Reply:
I used the wrong aspectRatio. Should be fixed now.